src/reader.rs

```diff
@@ -599,6 +604,102 @@ where
         Ok(Icon { href })
     }
 
+    fn read_link_type_icon(
+        &mut self,
+        attrs: HashMap<String, String>,
+    ) -> Result<LinkTypeIcon, Error> {
+        let mut icon = LinkTypeIcon {
+            attrs,
+            ..Default::default()
+        };
+        loop {
+            let mut e = self.reader.read_event(&mut self.buf)?;
+            match e {
+                Event::Start(ref mut e) => match e.local_name() {
+                    b"href" => icon.href = Some(self.read_str()?),
+                    b"refreshMode" => {
+                        icon.refresh_mode = match self.read_str()?.as_str() {
+                            "onChange" => Some(RefreshMode::OnChange),
+                            "onInterval" => Some(RefreshMode::OnInterval),
+                            "onExpire" => Some(RefreshMode::OnExpire),
+                            _ => None,
+                        }
+                    }
```

Comment:
Instead of including this here, could we implement `FromStr` for `RefreshMode` as well as the other added enums? We've done this for `AltitudeMode` and `ColorMode` as well and it keeps things a little cleaner

Reply:
Done.